htlcswitch/link.go

```diff
@@ -1414,6 +1418,14 @@ func (l *channelLink) handleDownStreamPkt(pkt *htlcPacket, isReProcess bool) {
 
 		l.cfg.Peer.SendMessage(false, htlc)
 
+		// Send a forward event notification to htlcNotifier.
+		l.cfg.HtlcNotifier.NotifyForwardingEvent(
+			NewForwardingEvent(
```

Comment:
It seems that you are using these constructors virtually everywhere. In that case, why not pass in the values to the notify methods directly? Also allows the timestamp to be set in the notifier itself.